sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/arithmetic.scala

```diff
@@ -394,25 +397,47 @@ case class Divide(left: Expression, right: Expression) extends DivModLike {
 // scalastyle:on line.size.limit
 case class IntegralDivide(left: Expression, right: Expression) extends DivModLike {
 
-  override def inputType: AbstractDataType = IntegralType
+  override def inputType: AbstractDataType = TypeCollection(IntegralType, DecimalType)
+
```

Comment:
(This comment is not related to this pr though...) We need to update the description for IntegralDivide: https://github.com/apache/spark/pull/25136/files#diff-1516b10738479bbe190fb4e239258473R390